src/main/java/uk/gov/hmcts/reform/userprofileapi/infrastructure/clients/CreateUserProfileData.java

```diff
@@ -1,38 +1,81 @@
 package uk.gov.hmcts.reform.userprofileapi.infrastructure.clients;
 
 import javax.validation.constraints.NotNull;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.NoArgsConstructor;
 import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.userprofileapi.domain.RequiredFieldMissingException;
 
+@Builder
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@NoArgsConstructor
 @Component
 public class CreateUserProfileData implements RequestData {
 
-    @NotNull
-    private String email;
+    @NotNull private String email;
+    @NotNull private String firstName;
+    @NotNull private String lastName;
+    private String languagePreference;
 
-    @NotNull
-    private String firstName;
+    private boolean emailCommsConsent;
+    private boolean postalCommsConsent;
 
-    @NotNull
-    private String lastName;
-
-    public CreateUserProfileData() {
-    }
-
-    public CreateUserProfileData(String email, String firstName, String lastName) {
-        this.email = email;
-        this.firstName = firstName;
-        this.lastName = lastName;
-    }
+    @NotNull private String userCategory;
+    @NotNull private String userType;
+    private String idamRoles;
 
     public String getEmail() {
+        if (email == null) {
+            throw new RequiredFieldMissingException("email must not be null");
+        }
```

Comment:
Couldn't an instance of this object be created in a way that ensures its mandatory values are populated?

Reply:
Hmm.  Because its initiated by the deserializer I thought it was possible it could be null.  Although there could be a way to do it I've not thought of so will have a think.

Reply:
will change this so that we don't have a no-args constructor for the deserializer and use the @JsonCreator annotation to specify to Jackson on how to create the object.  This should mean we can check for mandatory nulls in the constructor at creation time which is much better.